### ISSUE TITLE
[Performance] Hitching when SmallTime is loaded

### ISSUE THREAD

**Issue by grankless** [unsoluble/smalltime#106]
System: PF2e
This issue began on 8.9, but has persisted through v9. Me and my players have significant frame hitching when SmallTime is active, even with no other modules active. The new FPS meter in Foundry does show our FPS is reduced from ~67 to ~58-59, which is fine, but the hitching does not seem to be reflected in the FPS meter at all. It just does it.

Unsure of what other info I can provide.

**Comment by unsoluble** [comment]
Can you describe this "hitching" effect in more detail? What parts of the UI or play experience seem affected?

Is this perhaps only when the real-time clock is running?

Do you have PF2's darkness sync enabled?

**Comment by grankless** [comment]
The "camera" so to speak seems to freeze - here's a gif I screen captured. https://i.imgur.com/apCil8Q.gif 

It only happens when the game is unpaused, so the thing is running.

Synchronize Darkness is unselected.

**Comment by grankless** [comment]
We all have WebGL enabled ,I've verified. I will note that the performance issues are more severe on FIrefox than Chromium browsers.

**Comment by unsoluble** [comment]
Okay, thanks for the details.

Does it happen when the game is unpaused but the realtime clock is stopped? Or only when the clock is also running?

**Comment by grankless** [comment]
I didn't even think to do that - that DID stop it. Performance seems great with realtime clock stopped (through simplecalendar, not sure about how to do it with worldclock).

So it seems to only be with the clock running. I have verified with a player that it seems to have solved it on their end too. They seem to still be having a couple microstutters, but nothing that's at all a downside. Thank you for solving the issue for me as a stopgap - I have no idea if this will be like, a "fixable" issue, but at least I'm a satisfied customer. Thank you.

**Comment by unsoluble** [comment]
Hey thanks for poking at it for me -- _really_ helps to narrow things down like that.

I think probably it's just a matter of the constant clock updates rendering more frequently than they need to. Will investigate when I've got a chance here.

**Comment by grankless** [comment]
Thank you.

**Comment by grankless** [comment]
Oh hey, just saw this. I went in and did Find The Culprit again - I did find that having real time clock stuff on with Simple Calendar did reduce my FPS down to about 30...

Did Find the Culprit...

And then it was just fixed? I have no idea what happened. Maybe a setting got refreshed or something? But I finally have it resolved in at least one world, which is the one it was most markedly severe in. I'll try in other worlds as soon as I can.
